doc/devel/coding_guide.rst

```diff
@@ -13,9 +13,89 @@
 Pull request guidelines
 ***********************
 
-Pull requests (PRs) are the mechanism for contributing to Matplotlibs code and
+Pull requests (PRs) are the mechanism for contributing to Matplotlib's code and
 documentation.
 
+It is recommended to check that your contribution complies with the following
+rules before submitting a pull request:
+
+* If your pull request addresses an issue, please use the title to describe the
+  issue and mention the issue number in the pull request description to ensure
+  that a link is created to the original issue.
+
+* All public methods should have informative docstrings with sample usage when
+  appropriate. Use the `numpy docstring standard
+  <https://numpydoc.readthedocs.io/en/latest/format.html>`_.
+
+* Formatting should follow the recommendations of PEP8_, as enforced by
+  flake8_.  You can check flake8 compliance from the command line with ::
+
+    python -m pip install flake8
+    flake8 /path/to/module.py
+
+  or your editor may provide integration with it.  Note that Matplotlib
+  intentionally does not use the black_ auto-formatter (1__), in particular due
+  to its unability to understand the semantics of mathematical expressions
+  (2__, 3__).
+
+  .. _PEP8: https://www.python.org/dev/peps/pep-0008/
+  .. _flake8: https://flake8.pycqa.org/
+  .. _black: https://black.readthedocs.io/
+  .. __: https://github.com/matplotlib/matplotlib/issues/18796
+  .. __: https://github.com/psf/black/issues/148
+  .. __: https://github.com/psf/black/issues/1984
+
+* Each high-level plotting function should have a simple example in the
+  ``Example`` section of the docstring.  This should be as simple as possible
```

Comment:
I suggest we drop this. We don't consistently have this in our current docs.

 Examples calls are usually not very instuctive without seeing the visual output. And we have the sphinx-gallery mini galleries at the bottom of the docstring.


We don't consistently have this in our current docs, and I 

If we want this, we should move it to https://matplotlib.org/stable/devel/documenting_mpl.html#writing-docstrings

Reply:
I think you're right - and this is actually pretty well explained already in that page. Let me know if what I wrote makes sense, or if we should remove entirely.